go-controller/pkg/config/config_test.go

```diff
@@ -1082,6 +1090,46 @@ enable-pprof=true
 		err := app.Run(cliArgs)
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 	})
+	It("returns an error when the v4 masquerade subnet specified is invalid", func() {
+		app.Action = func(ctx *cli.Context) error {
+			_, err := InitConfig(ctx, kexec.New(), nil)
+			gomega.Expect(err).To(gomega.MatchError("invalid gateway v4 masquerade subnet specified, subnet: foobar: error: invalid CIDR address: foobar"))
+			return nil
+		}
+		cliArgs := []string{
+			app.Name,
+			"-gateway-v4-masquerade-subnet=foobar",
+		}
+		err := app.Run(cliArgs)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	})
+	It("returns an error when the v6 masquerade subnet specified is invalid", func() {
+		app.Action = func(ctx *cli.Context) error {
+			_, err := InitConfig(ctx, kexec.New(), nil)
+			gomega.Expect(err).To(gomega.MatchError("invalid gateway v6 masquerade subnet specified, subnet: 192.168.0.0/16: error: <nil>"))
+			return nil
+		}
+		cliArgs := []string{
+			app.Name,
+			"-gateway-v6-masquerade-subnet=192.168.0.0/16",
+		}
+		err := app.Run(cliArgs)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	})
+	It("successfully overrides the default masquerade subnets", func() {
+		app.Action = func(ctx *cli.Context) error {
+			_, err := InitConfig(ctx, kexec.New(), nil)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			return nil
+		}
+		cliArgs := []string{
+			app.Name,
+			"-gateway-v4-masquerade-subnet=186.188.147.0/16",
+			"-gateway-v6-masquerade-subnet=fd68::/125",
+		}
+		err := app.Run(cliArgs)
```

Comment:
can we add verification here that the config.Gateway.MasqueradeIPs values are what we expect after overriding?